src/Symfony/Component/Routing/Loader/ControllerAwareXmlFileLoader.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+use Symfony\Component\Config\FileLocatorInterface;
+
+/**
+ * Adds support for a "controller" attribute.
+ *
+ * @author Oleg Voronkovich <oleg-voronkovich@yandex.ru>
+ */
+class ControllerAwareXmlFileLoader extends XmlFileLoader
```

Comment:
I asked myself the same. But I came to the conclusion it's legit, as the controller parameter is "tied" to the HttpKernel usage of the Routing component.
But then, I wonder if this rather belong to the `HttpKernel` component or the `FrameworkBundle` instead?